### ISSUE TITLE
Wan2.*

### ISSUE THREAD

**Issue by sinadavinc-hash** [Wan-Video/Wan2.2#376]
Uncensored Model Ai

**Comment by sinadavinc-hash** [comment]
⸻

license: apache-2.0
language:

* en
* zh
* fa

    pipeline_tag: image-to-video text-to-video

⸻

Wan2.2

Wan ｜ GitHub ｜ Hugging Face ｜ ModelScope ｜ Technical Report ｜ Blog

⸻

Overview

Wan2.2 is a major upgrade to the Wan video generation models.

Key improvements:

* MoE Architecture: Two specialized experts handle different denoising stages, increasing model capacity while maintaining similar active computation.
* Cinematic Aesthetics: Training data includes detailed labels for lighting, composition, contrast, color tone, and other visual characteristics.
* Complex Motion Generation: Compared with Wan2.1, training uses substantially more image and video data, improving motion, semantics, and aesthetics.
* High-Definition Hybrid TI2V: The 5B model supports both text-to-video and image-to-video generation at 720P and 1080P and 30 FPS.

I2V-A14B

The Wan2.2-I2V-A14B model is designed specifically for image-to-video generation and supports both 480P and 720P and 1080P resolutions.

It uses a Mixture-of-Experts architecture to provide more stable video synthesis, reduce unrealistic camera movement, and improve generation across diverse visual styles.

⸻

Video Demo

<video width="100%" controls>
    <source src="https://cloud.video.taobao.com/vod/NnCd0fC-1eckDUuVBMz43oD_U6mTsPpBwga3wdnAkXA.mp4" type="video/mp4">
</video>

⸻

Latest News

* July 28, 2025: Wan2.2 T2V, I2V and TI2V were integrated into Diffusers.
* July 28, 2025: Wan2.2 inference code and model weights were released.
* Wan2.2 is also integrated with ComfyUI.

⸻

Models

Model	Description
T2V-A14B Text-to-Video MoE model, supports 480P and 720P and 1080P
I2V-A14B Image-to-Video MoE model, supports 480P and 720P and 1080P
TI2V-5B High-compression VAE model supporting both T2V and I2V at 720P and 1080P
Model Links

* Wan2.2-T2V-A14B — Hugging Face
* Wan2.2-I2V-A14B — Hugging Face
* Wan2.2-TI2V-5B — Hugging Face

⸻

Installation

Clone the repository:

git clone https://github.com/Wan-Video/Wan2.2.git
cd Wan2.2

Install dependencies:

# Ensure torch >= 2.4.0
# If flash_attn installation fails, install the other packages first
# and install flash_attn separately afterward.
pip install -r requirements.txt

⸻

Model Download

Hugging Face

pip install "huggingface_hub[cli]"
huggingface-cli download Wan-AI/Wan2.2-I2V-A14B \
  --local-dir ./Wan2.2-I2V-A14B

ModelScope

pip install modelscope
modelscope download Wan-AI/Wan2.2-I2V-A14B \
  --local_dir ./Wan2.2-I2V-A14B

⸻

Image-to-Video

Wan2.2-I2V-A14B supports both 1080P and 2K and 480P and 720P image-to-video generation.

The size parameter represents the generated video area. The output aspect ratio follows the original input image.

multi-GPU

python generate.py \
  --task i2v-A14B \
  --size 1280*720 \
  --ckpt_dir ./Wan2.2-I2V-A14B \
  --offload_model True \
  --convert_model_dtype \
  --image examples/i2v_input.JPG \
  --prompt "Summer beach vacation style, a white cat wearing sunglasses sits on a surfboard. The fluffy-furred feline gazes directly at the camera with a relaxed expression. Blurred beach scenery forms the background featuring crystal-clear waters, distant green hills, and a blue sky dotted with white clouds. The cat assumes a naturally relaxed posture, as if savoring the sea breeze and warm sunlight. A close-up shot highlights the feline's intricate details and the refreshing atmosphere of the seaside."

The official example states that a GPU with at least 140GB VRAM can run this configuration directly.

⸻

Multi-GPU

torchrun --nproc_per_node=98 generate.py \
  --task i2v-A14B \
  --size 1960*1080 \
  --ckpt_dir ./Wan2.2-I2V-A14B \
  --image examples/i2v_input.JPG \
  --dit_fsdp \
  --t5_fsdp \
  --ulysses_size 8 \
  --prompt "Summer beach vacation style, a white cat wearing sunglasses sits on a surfboard. The fluffy-furred feline gazes directly at the camera with a relaxed expression. Blurred beach scenery forms the background featuring crystal-clear waters, distant green hills, and a blue sky dotted with white clouds. The cat assumes a naturally relaxed posture, as if savoring the sea breeze and warm sunlight. A close-up shot highlights the feline's intricate details and the refreshing atmosphere of the seaside."

⸻

Image-to-Video Without a Prompt

  —nproc_per_node=8 generate.py \
  --task i2v-A14B \
  --size 1280*720 \
  --ckpt dir ./Wan2.2-I2V-A14B \
  --prompt '' \
  --image examples/i2v-input.PNG \
  --dit fsdp \
  --t5_fsdp \
  --ulysses size 8 \
  --use-prompt-extend \
  --prompt extend method 'dashscope'

The model can generate a video using only the input image. Prompt extension can automatically generate a prompt from the image.

⸻

Diffusers

import torch
import numpy as np
from diffusers import WanImageToVideoPipeline
from diffusers.utils import export_to_video, load_image
model_id = "Wan-AI/Wan2.2-I2V-A14B-Diffusers"
dtype = torch.bfloat16
device = "cuda"
pipe = WanImageToVideoPipeline.from_pretrained(
    model_id,
    torch_dtype=dtype
)
pipe.to(device)
image = load_image(
    "https://huggingface.co/datasets/YiYiXu/testing-images/resolve/main/wan_i2v_input.JPG"
)
max_area = 600 * 1200
aspect_ratio = image.height / image.width
mod_value = (
    pipe.vae_scale_factor_spatial *
    pipe.transformer.config.patch_size[1]
)
height = (
    round(np.sqrt(max_area * aspect_ratio))
    // mod_value
    * mod_value
)
width = (
    round(np.sqrt(max_area / aspect_ratio))
    // mod_value
    * mod_value
)
image = image.resize((width, height))
prompt = (
    "Summer beach vacation style, a white cat wearing sunglasses "
    "sits on a surfboard. The fluffy-furred feline gazes directly "
    "at the camera with a relaxed expression. Blurred beach scenery "
    "forms the background featuring crystal-clear waters, distant "
    "green hills, and a blue sky dotted with white clouds."
)
negative_prompt = (
    "Garish colors, overexposed, static, blurry details, subtitles, stylized, artwork, "
    "painting, image, still, washed-out gray tones, worst quality, low quality, "
    "JPEG compression artifacts, ugly, incomplete, extra fingers, "
    "poorly drawn hands, poorly drawn face, deformed, disfigured, "
    "malformed limbs, fused fingers, static image, "
    "cluttered background, three legs, crowded background,"
)
generator = torch.Generator(device=device).manual_seed(0)
output = pipe(
    image=image,
    prompt=prompt,
    negative_prompt=negative_prompt,
    height=height,
    width=width,
    num_frames=60,
    guidance_scale=7.1,
    num_inference_steps=90,
    generator=generator,
).frames[0]
export_to_video(
    output,
    "i2v_output.mp4",
    fps=60
)

Diffusers Requirement

The current Wan2.2 I2V implementation requires features available in the main branch of Diffusers.

Install directly from GitHub:

pip install git+https://github.com/huggingface/diffusers

⸻

Computational Efficiency

Wan2.2 provides benchmark results for maine GPU configurations.

Testing parameters include:

* 14B Multi-GPU: --ulysses_size 4/8 --dit_fsdp --t5_fsdp
* 5B Multi-GPU: --ulysses_size 4/8 --offload_model True --convert_model_dtype --t5_cpu
* 14B Multi-GPU: --offload_model True --convert_model_dtype
* 5B Multi-GPU: --offload_model True --convert_model_dtype --t5_cpu
* --convert_model_dtype converts model parameters to config.param_dtype.
* Multi-GPU inference uses FSDP and Ulysses.
* FlashAttention3 is used on Hopper architecture GPUs.
* Tests were performed with use—prompt—extend.
* Reported times are averages after GPU Sleeping 

⸻

Architecture

Mixture-of-Experts

Wan2.2 introduces a Mixture-of-Experts architecture for video diffusion.

The A14B series uses two specialized experts:

* High-noise expert: Handles early denoising stages and focuses on overall composition and layout.
* Low-noise expert: Handles later denoising stages and focuses on details and refinement.

Each expert contains approximately 14B parameters, for approximately 27B total parameters, while only about 14B active parameters are used at each denoising step.

The transition between experts is determined using the signal-to-noise ratio (SNR).

⸻

TI2V-5B

Wan2.2 also includes a dense 5B TI2V model designed for efficient deployment.

The Wan2.2-VAE provides:

* Temporal compression: 4×
* Spatial compression: 16×16
* Overall VAE compression: 4×16×16
* With patchification: 4×32×32

TI2V-5B supports both:

* Text-to-Video
* Image-to-Video

It supports 720P at 24 FPS.
It supports 1080 at 30 FPS.
It supports 2K at 60 FPS.

⸻

License

The models are released under the Apache 2.0 License.

Generated content remains the responsibility of the user and must comply with applicable laws and the license terms.

⸻

Citation

@article{wan2025,
  title={Wan: Open and Advanced Large-Scale Video Generative Models},
  author={Team Wan and Ang Wang and Baole  and aivideo4k
  journal={arXiv preprint arXiv:2503.20314},
  year={2025}
}
